node/src/components/contract_runtime.rs

```diff
@@ -168,36 +169,48 @@ where
                 chainspec,
                 responder,
             } => {
-                let result = self.commit_genesis(&chainspec);
-                if let Err(lmdb_error) = self.engine_state.flush_environment() {
-                    fatal!(
-                        effect_builder,
-                        "error flushing lmdb environment {:?}",
-                        lmdb_error
-                    )
-                    .ignore()
-                } else {
-                    trace!(?result, "commit_genesis response");
-                    responder.respond(result).ignore()
+                let engine_state = Arc::clone(&self.engine_state);
+                async move {
+                    let result = commit_genesis(&chainspec, &engine_state);
+                    if let Err(lmdb_error) = engine_state.flush_environment() {
+                        fatal!(
+                            effect_builder,
+                            "error flushing lmdb environment {:?}",
+                            lmdb_error
+                        )
+                        .await;
+                    } else {
+                        trace!(?result, "commit_genesis response");
+                        responder.respond(result).await;
+                    }
```

Comment:
I'm not understanding why this change was made - just want to check I'm not missing some important detail.  Same goes for the upgrade request.